src/router/swap.ts

```diff
@@ -71,7 +71,10 @@ export abstract class TokenSwap {
                 routeOutput.route.tokenIn,
                 routeOutput.route.tokenOut,
                 tokenInAmount,
-                minTokenOutAmount
+                minTokenOutAmount,
+                {
+                    value: routeOutput.nativeSend ? tokenInAmount : 0,
```

Comment:
routeOutput.nativeSend is an array of boolean. Need to check 0th entry imo. But recheck.